.github/scripts/run-container-test.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Run precompiled Kotlin/Native test binaries in a Docker container for a specific Linux distribution and architecture.
+
+This requires Docker multiarch support, see https://docs.docker.com/build/building/multi-platform/ and https://github.com/multiarch/qemu-user-static
+In GitHub we use a provided action for this: https://github.com/docker/setup-qemu-action
+
+Locally you would need to run one of:
+
+`docker run --rm --privileged multiarch/qemu-user-static --reset -p yes --credential yes`
+
+OR
+
+`docker run --privileged --rm tonistiigi/binfmt --install all`
+"""
+
+import argparse
+import os
+import subprocess
+import shlex
+import shutil
+
+VERBOSE = False
+
+DISTRO_TO_IMAGE_NAME= {
+    "ubuntu-22.04": "public.ecr.aws/lts/ubuntu:22.04_stable",
+    "al2023": "public.ecr.aws/amazonlinux/amazonlinux:2023",
+    "al2": "public.ecr.aws/amazonlinux/amazonlinux:2"
```

Comment:
Should we pin on specific versions of these images? 

- `public.ecr.aws/amazonlinux/amazonlinux:2023.3.20240205.2`
- `public.ecr.aws/amazonlinux/amazonlinux:2.0.20240131.0`

Reply:
I went back and forth on this. On one hand yes that makes total sense on the other taking the latest ensures we are continue to be compatible and are alerted of any new issues. 

We _could_ do both I suppose, pin to one and have a version flag or something. I opted to just take latest at the moment but I'm interested in what people think here.